nixos/modules/system/boot/stage-1-init.sh

```diff
@@ -152,22 +155,29 @@ exec > /tmp/stage-1-init.log.fifo 2>&1
 # Process the kernel command line.
 export stage2Init=/init
 for o in $(cat /proc/cmdline); do
+    # fixing SC2046/SC2086 (quote vars) breaks with
```

Comment:
You can treat the command line as an array by using the `mapfile` (alias `readarray`) builtin:

```
$ diff --ignore-blank-lines <(cat /proc/cmdline) <(mapfile args < /proc/cmdline && echo "${args[@]}")
$ echo $?
0
```